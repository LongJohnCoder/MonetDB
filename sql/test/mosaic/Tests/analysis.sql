START TRANSACTION;

create table if not exists tmp4 (i int);

truncate TABLE tmp4;

insert into tmp4 select * from generate_series(0, 10000000);

-- tmp4 is currently uncompressed

select technique, factor from mosaic.analysis('sys', 'tmp4', 'i') order by factor desc, technique;

-- should be materialized as the graph of a cutoff function.

insert into tmp4 select 10000000 from tmp4;

select technique, factor from mosaic.analysis('sys', 'tmp4', 'i') order by factor desc, technique;

select technique, factor from mosaic.analysis('sys', 'tmp4', 'i', 'linear, runlength') order by factor desc, technique;

-- Make sure that mosaic.analysis also works correctly on a column with compression

set optimizer='mosaic_pipe';

alter table tmp4 alter column i set storage 'dict';

select technique, factor from mosaic.analysis('sys', 'tmp4', 'i') order by factor desc, technique;

select technique, factor from mosaic.analysis('sys', 'tmp4', 'i', 'linear, runlength') order by factor desc, technique;

drop table tmp4;

ROLLBACK;
