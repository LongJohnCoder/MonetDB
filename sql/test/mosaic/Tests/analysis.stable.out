stdout of test 'analysis` in directory 'sql/test/mosaic` itself:


# 16:07:15 >  
# 16:07:15 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=38225" "--set" "mapi_usock=/var/tmp/mtest-15728/.s.monetdb.38225" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/aris/sources/monetdb/build/installation/var/MonetDB/mTests_sql_test_mosaic" "--set" "monet_daemon=yes" "--set" "embedded_c=true"
# 16:07:15 >  

# MonetDB 5 server v11.34.0 (hg id: 4fe7de431d91+)
# This is an unreleased version
# Serving database 'mTests_sql_test_mosaic', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.349 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://aris-XPS-13-9380:38225/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-15728/.s.monetdb.38225
# MonetDB/SQL module loaded

# 16:07:15 >  
# 16:07:15 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-15728" "--port=38225"
# 16:07:15 >  

#create table tmp4 (i int);
#insert into tmp4 select * from generate_series(0,10000000);
[ 10000000	]
#select technique, factor from mosaic.analysis('sys', 'tmp4', 'i') order by factor desc, technique;
% .L4,	.L4 # table_name
% technique,	factor # name
% clob,	double # type
% 13,	24 # length
[ "linear ",	2886.002	]
[ "frame ",	2449.179	]
[ "delta ",	3.994	]
[ "frame prefix ",	2.284	]
[ "prefix ",	2.284	]
[ "var ",	0.999	]
[ "runlength ",	0.999	]
#select technique, factor from mosaic.analysis('sys', 'tmp4', 'i', 'var, runlength') order by factor desc, technique;
% .L5,	.L5 # table_name
% technique,	factor # name
% clob,	double # type
% 11,	24 # length
[ "var ",	0.999	]
[ "runlength ",	0.999	]
#insert into tmp4 select 10000000 from tmp4;
[ 10000000	]
#select technique, factor from mosaic.analysis('sys', 'tmp4', 'i') order by factor desc, technique;
% .L4,	.L4 # table_name
% technique,	factor # name
% clob,	double # type
% 23,	24 # length
[ "runlength linear ",	5457.025	]
[ "linear ",	5312.084	]
[ "runlength frame ",	4669.624	]
[ "frame ",	4563.084	]
[ "delta frame ",	7.988	]
[ "runlength delta ",	7.988	]
[ "delta prefix ",	7.101	]
[ "runlength frame prefix ",	4.568	]
[ "runlength prefix ",	4.567	]
[ "frame prefix ",	4.266	]
[ "prefix ",	4.263	]
[ "delta ",	3.997	]
[ "runlength ",	1.999	]
[ "var ",	1.454	]
#select technique, factor from mosaic.analysis('sys', 'tmp4', 'i', 'linear, runlength') order by factor desc, technique;
% .L5,	.L5 # table_name
% technique,	factor # name
% clob,	double # type
% 17,	24 # length
[ "runlength linear ",	5457.025	]
[ "linear ",	5312.084	]
[ "runlength ",	1.999	]
#set optimizer='mosaic_pipe';
#alter table tmp4 alter column i set storage 'var';
#select technique, factor from mosaic.analysis('sys', 'tmp4', 'i', 'linear, runlength') order by factor desc, technique;
% .L5,	.L5 # table_name
% technique,	factor # name
% clob,	double # type
% 17,	24 # length
[ "runlength linear ",	5457.025	]
[ "linear ",	5312.084	]
[ "runlength ",	1.999	]
#drop table tmp4;

# 16:08:07 >  
# 16:08:07 >  "Done."
# 16:08:07 >  

